bcipy/simulator/helpers/sim_state.py

```diff
@@ -1,28 +1,129 @@
     target_sentence: str
-    display_alphabet: list[str]
+    display_alphabet: List[str]
     inquiry_n: int
     series_n: int
+    series_results: List[List[InquiryResult]]
+
+    decision_criterion: List[SimDecisionCriteria]
+
+    def total_inquiry_count(self):
+        count = 0
+        for series in self.series_results:
+            count += len(series)
+
+        return count
 
 
 class StateManager(ABC):
 
-    def update(self, evidence):
-        ...
+    def update(self, evidence: np.ndarray):
+        raise NotImplementedError()
 
     def is_done(self) -> bool:
-        ...
+        raise NotImplementedError()
 
-    def get_state(self) -> SimSessionState:
+    def get_state(self) -> SimState:
         ...
 
-    def add_state(self, state_field, state_value):
+    def mutate_state(self, state_field, state_value):
         pass
+
+
+class StateManagerImpl(StateManager):
+
+    def __init__(self, parameters: Parameters):
+        self.state: SimState = self.initial_state()
+        self.parameters = parameters
+
+        self.stop_inq = 50  # TODO pull from parameters
+
+    def is_done(self) -> bool:
+
+        return self.state.total_inquiry_count() > self.stop_inq or self.state.target_sentence == self.state.current_sentence or self.state.series_n > 50
+
+    def update(self, evidence: np.ndarray) -> InquiryResult:
+
+        temp_inquiry_result = InquiryResult(target=self.state.target_symbol, time_spent=0, stimuli=self.state.display_alphabet,
+                                            evidence_likelihoods=list(evidence), decision=None)
+
+        # finding out whether max iterations is hit or prob threshold is hit
+        temp_series = copy.deepcopy(self.get_state().series_results)
+        temp_series[-1].append(temp_inquiry_result)
+        is_decidable = any([decider.decide(temp_series[-1]) for decider in self.state.decision_criterion])
+        decision = None
+        # TODO what to do when max inquiry count is reached?
+
+        new_state = self.get_state().__dict__
+        if is_decidable:
+            decision = alphabet()[np.argmax(evidence)]  # deciding the maximum probability symbol TODO abstract
+
+            if decision == self.state.target_symbol:  # correct decision
+                new_state['series_n'] += 1  # TODO abstract out into reset function
+                new_state['series_results'].append([])
+                new_state['inquiry_n'] = 0
+
+                new_state['current_sentence'] += decision
+
+                next_target_symbol_idx = len(new_state['current_sentence'])
+                if next_target_symbol_idx >= len(self.state.target_sentence):
+                    pass
+                else:
+                    new_state['target_symbol'] = self.state.target_sentence[next_target_symbol_idx]
+
+            else:  # wrong decision
+                new_state['series_n'] += 1
+                new_state['series_results'].append([])
+                new_state['inquiry_n'] = 0
```

Comment:
In the case that the spelled letter does not match the target, it it still displayed as part of the current sentence. However, then the next target becomes the backspace character. See the `RSVPCopyPhraseTask` class `next_target` method. You probably don't need an if-else here. I think the only thing that would be different is the selection of the next target.